google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/standard/AppEngineStandardDeploymentEditor.java

```diff
@@ -0,0 +1,246 @@
+import com.intellij.openapi.project.Project;
+import com.intellij.ui.HyperlinkLabel;
+import com.intellij.ui.components.JBTextField;
+import com.intellij.util.ui.tree.TreeModelAdapter;
+
+import org.apache.commons.lang.StringUtils;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Set;
+
+import javax.swing.JCheckBox;
+import javax.swing.JComponent;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.event.TreeModelEvent;
+
+/**
+ * Editor for an App Engine Deployment runtime configuration.
+ */
+public class AppEngineStandardDeploymentEditor extends
+    SettingsEditor<AppEngineDeploymentConfiguration> {
+  private JPanel editorPanel;
+  private JBTextField versionIdField;
+  private ProjectSelector projectSelector;
+  private JCheckBox promoteCheckbox;
+  private JCheckBox stopPreviousVersionCheckbox;
+  private HyperlinkLabel promoteInfoLabel;
+  private AppEngineApplicationInfoPanel applicationInfoPanel;
+  private JLabel serviceLabel;
+  private JLabel environmentLabel;
+  private HyperlinkLabel appEngineCostWarningLabel;
+
+  private Project project;
+  private AppEngineDeployable deploymentSource;
+
+  private static final boolean PROMOTE_DEFAULT = true;
+  private static final boolean STOP_PREVIOUS_VERSION_DEFAULT = true;
+
+  /**
+   * Initializes the UI components.
+   */
+  public AppEngineStandardDeploymentEditor(Project project,
+      final AppEngineDeployable deploymentSource) {
+    this.project = project;
+    this.deploymentSource = deploymentSource;
+
+    versionIdField.getEmptyText().setText(
+        GctBundle.message("appengine.flex.version.placeholder.text"));
+    promoteCheckbox.setSelected(PROMOTE_DEFAULT);
+    stopPreviousVersionCheckbox.setVisible(
+        AppEngineProjectService.getInstance().isFlexCompat(project, deploymentSource)
+    );
+    stopPreviousVersionCheckbox.setSelected(STOP_PREVIOUS_VERSION_DEFAULT);
+
+    promoteInfoLabel.setHyperlinkText(
+        GctBundle.getString("appengine.promote.info.label.beforeLink") + " ",
+        GctBundle.getString("appengine.promote.info.label.link"),
+        "");
+    promoteInfoLabel.addHyperlinkListener(new BrowserOpeningHyperLinkListener());
+    promoteInfoLabel.setHyperlinkTarget(GctBundle.getString("appengine.promoteinfo.url"));
+
+    promoteCheckbox.addItemListener(event -> {
+      boolean isPromoteSelected = ((JCheckBox) event.getItem()).isSelected();
+
+      stopPreviousVersionCheckbox.setEnabled(isPromoteSelected);
+
+      if (!isPromoteSelected) {
+        stopPreviousVersionCheckbox.setSelected(false);
+      }
+    });
+
+    projectSelector.addProjectSelectionListener(applicationInfoPanel::refresh);
+
+    projectSelector.addModelListener(new TreeModelAdapter() {
+      @Override
+      public void treeStructureChanged(TreeModelEvent event) {
+        // projects have finished loading
+        refreshApplicationInfoPanel();
+      }
+    });
+
+    serviceLabel.setText(AppEngineProjectService.getInstance()
```

Comment:
Actually, this is not a regression -- it's already happening in master. To reproduce, create a Standard module, fill in a deployment configuration and then add another Standard module to the project.
The existing run configuration is overriden with a blank one.

The issue is that we're adding configurations with the same name, and IJ isn't smart enough to create them on the side. In a previous version of this branch, I was appending the module name to the run configuration and different run configs were created.

Since it's not a regression, my suggestion is to file an issue and handle later. Let me know what you think.